design/plankton/CommunicationLayerDesign.md

```diff
@@ -36,13 +36,161 @@ The IBEX team at ISIS, for example, have expressed interest in using Plankton. N
 
 ## Design Options
 
-### Option A
+### Adapter-Binder Approach
 
-Detail design option A here.
+Overview:
+
+* `Adapters` are bespoke per `Device` as well as per `Protocol`.
+* We term the `Hardware` as per the manufacturer specification to be `Device` + `Adapter`.
+* There is no separate binding.
+* There is no general concrete `Adapter` for any protocol. 
+* The process loop is kicked off by the `Environment`
+
+#### Adapters
+
+The problem with the current implementation is that the `Adapter` concept is too general and the `Binder` mechanism is too limited (1:1) to make linkup to a `Device` powerful or flexible enough.
+
+The `Adapters` in this design are per-protocol & per-device. So they can encapsulate all the logic necessary to drive the specific device that they wrap. 
+
+```python
+class StreamAdapterLinkum95(StreamAdapter):
```

Comment:
@MichaelWedel Ok. I didn't have much insight into how the base classes could be designed but it would seem sensible to avoid excessive duplication where we need utilities for a particular protocol. I think the base class has two purposes.

1. To avoid code duplication by increasing sharing, which improves maintainability and should make adapter implementations easier.
2. For introspection so that for a given `Device` we can determine what `Adapter(s)` have been made available at ArgParser time.